src/main/java/org/json/JSONObject.java

```diff
@@ -208,16 +208,16 @@ public JSONObject(JSONObject jo, String ... names) {
     }
 
     /**
-     * Construct a JSONObject from a JSONTokener.
-     *
-     * @param x
-     *            A JSONTokener object containing the source string.
-     * @throws JSONException
-     *             If there is a syntax error in the source string or a
-     *             duplicated key.
-     */
+         * Construct a JSONObject from a JSONTokener.
+         *
+         * @param x
+         *            A JSONTokener object containing the source string.
+         * @throws JSONException
+         *             If there is a syntax error in the source string or a
+         *             duplicated key.
+         */
     public JSONObject(JSONTokener x) throws JSONException {
-        this();
+        this(x.getMapImplementation());
```

Comment:
Better do a null check before invoking on that instance

Reply:
The tokener is expected to be non-null always there, the the called constructor checks for a null parameter already. This looks fine as-is, but possibly ensure the java doc calls out the expectation.